webrtc-c/canary/src/Peer.cpp

```diff
@@ -533,13 +555,145 @@ STATUS Peer::writeFrame(PFrame pFrame, MEDIA_STREAM_TRACK_KIND kind)
+    DLOGD("duration:%lf", currentDuration);
+    this->canaryIncomingRTPMetricsContext.packetReceiveRate =
+        (DOUBLE)(this->canaryMetrics.rtcStatsObject.inboundRtpStreamStats.received.packetsReceived -
+                 this->canaryIncomingRTPMetricsContext.prevPacketsReceived) /
+        currentDuration;
+    this->canaryIncomingRTPMetricsContext.incomingBitRate =
+        ((DOUBLE)(this->canaryMetrics.rtcStatsObject.inboundRtpStreamStats.bytesReceived - this->canaryIncomingRTPMetricsContext.prevBytesReceived) /
+        currentDuration) / 0.008;
+    this->canaryIncomingRTPMetricsContext.framesDroppedPerSecond =
+        ((DOUBLE) this->canaryMetrics.rtcStatsObject.inboundRtpStreamStats.received.framesDropped -
+         this->canaryIncomingRTPMetricsContext.prevFramesDropped) /
+        currentDuration;
+
+    this->canaryIncomingRTPMetricsContext.prevPacketsReceived = this->canaryMetrics.rtcStatsObject.inboundRtpStreamStats.received.packetsReceived;
+    this->canaryIncomingRTPMetricsContext.prevBytesReceived = this->canaryMetrics.rtcStatsObject.inboundRtpStreamStats.bytesReceived;
+    this->canaryIncomingRTPMetricsContext.prevFramesDropped = this->canaryMetrics.rtcStatsObject.inboundRtpStreamStats.received.framesDropped;
+    this->canaryIncomingRTPMetricsContext.prevTs = this->canaryMetrics.timestamp;
+
+    DLOGD("Packet receive rate: %lf", this->canaryIncomingRTPMetricsContext.packetReceiveRate);
+    DLOGD("Incoming bit rate: %lf", this->canaryIncomingRTPMetricsContext.incomingBitRate / 1024.0);
+    DLOGD("Frame drop rate: %lf", this->canaryIncomingRTPMetricsContext.framesDroppedPerSecond);
+CleanUp:
+    return retStatus;
+}
+STATUS Peer::publishStatsForCanary()
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    std::lock_guard<std::recursive_mutex> lock(this->mutex);
+    switch (this->canaryMetrics.requestedTypeOfStats) {
+        case RTC_STATS_TYPE_OUTBOUND_RTP:
+            CHK_LOG_ERR(::rtcPeerConnectionGetMetrics(this->pPeerConnection, this->videoTransceivers.back(), &this->canaryMetrics));
+            this->populateOutgoingRtpMetricsContext();
+            Canary::Cloudwatch::getInstance().monitoring.pushOutboundRtpStats(&this->canaryOutgoingRTPMetricsContext);
+            break;
+        case RTC_STATS_TYPE_INBOUND_RTP:
+            CHK_LOG_ERR(::rtcPeerConnectionGetMetrics(this->pPeerConnection, this->videoTransceivers.back(), &this->canaryMetrics));
+            this->populateIncomingRtpMetricsContext();
+            Canary::Cloudwatch::getInstance().monitoring.pushInboundRtpStats(&this->canaryIncomingRTPMetricsContext);
+
+            break;
+        default:
+            CHK(FALSE, STATUS_NOT_IMPLEMENTED);
+    }
 CleanUp:
+    return retStatus;
+}
 
+STATUS Peer::publishEndToEndMetrics()
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    std::lock_guard<std::recursive_mutex> lock(this->mutex);
```

Comment:
Same here

Reply:
Actually, now that I think about I dont need it. Removing.